File: Source/Fastore.Client/Dictionary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Alphora.Fastore.Client
{
	/// <summary> The column IDs for the core and the client. </summary>
	/// <remarks> ColumnIDs 0-9,999 are reserved for the core engine.  Column IDs 10,000 through 
	/// 19,999 are reserved for the client. </remarks>
	public static class Dictionary
	{
		public const int MaxSystemColumnID = 9999;
		public const int MaxClientColumnID = 19999;

		public const int ColumnID = 0;
		public const int ColumnName = 1;
		public const int ColumnValueType = 2;
		public const int ColumnRowIDType = 3;
		public const int ColumnBufferType = 4;

		public static readonly int[] ColumnColumns = 
		{ 
			ColumnID,
			ColumnName,
			ColumnValueType,
			ColumnRowIDType,
			ColumnBufferType
		};

		public const int TopologyID = 100;

		public static readonly int[] TopologyColumns = 
		{ 
			TopologyID
		};

		public const int HostID = 200;

		public static readonly int[] HostColumns =
		{
			HostID
		};

		public const int PodID = 300;
		public const int PodHostID = 301;

		public static readonly int[] TablePodColumns = 
		{ 
			PodID,
			PodHostID
		};

		public const int PodColumnPodID = 400;
		public const int PodColumnColumnID = 401;

		public static readonly int[] PodColumnColumns = 
		{ 
			PodColumnPodID,
			PodColumnColumnID
		};

		public const int GeneratorNextValue = 10000;

		public static readonly int[] GeneratorColumns = 
		{ 
			GeneratorNextValue 
		};
	}
}

File: Source/FastoreCommon/Schema/Dictionary.cpp
#include "Dictionary.h"

#define COUNTOF(a) (sizeof(a)/sizeof(*a))

using namespace fastore::common;
using fastore::communication::ColumnID;

const ColumnID Dictionary::MaxSystemColumnID(9999);
const ColumnID Dictionary::MaxClientColumnID(19999);
const ColumnID Dictionary::ColumnID(0);
const ColumnID Dictionary::ColumnName(1);
const ColumnID Dictionary::ColumnValueType(2);
const ColumnID Dictionary::ColumnRowIDType(3);
const ColumnID Dictionary::ColumnBufferType(4);	
const ColumnID Dictionary::ColumnRequired(5);
const ColumnID Dictionary::TopologyID(100);
const ColumnID Dictionary::HostID(200);
const ColumnID Dictionary::PodID(300);
const ColumnID Dictionary::PodHostID(301);
const ColumnID Dictionary::PodColumnPodID(400);
const ColumnID Dictionary::PodColumnColumnID(401);
const ColumnID Dictionary::StashID(500);
const ColumnID Dictionary::StashHostID(501);
const ColumnID Dictionary::StashColumnStashID(600);
const ColumnID Dictionary::StashColumnColumnID(601);


const ColumnID Dictionary::_ColumnColumns[] = {ColumnID, ColumnName, ColumnValueType, ColumnRowIDType, ColumnBufferType, ColumnRequired };
const ColumnIDs  Dictionary::ColumnColumns(_ColumnColumns, _ColumnColumns + COUNTOF(_ColumnColumns));

const ColumnID Dictionary::_TopologyColumns[] = {TopologyID};
const ColumnIDs  Dictionary::TopologyColumns(_TopologyColumns, _TopologyColumns + COUNTOF(_TopologyColumns));

const ColumnID Dictionary::_HostColumns[] = {HostID};
const ColumnIDs  Dictionary::HostColumns(_HostColumns, _HostColumns + COUNTOF(_HostColumns));

const ColumnID Dictionary::_TablePodColumns[] = {PodID, PodHostID};
const ColumnIDs  Dictionary::TablePodColumns(_TablePodColumns, _TablePodColumns + COUNTOF(_TablePodColumns));

const ColumnID Dictionary::_PodColumnColumns[] = {PodColumnPodID, PodColumnColumnID};
const ColumnIDs  Dictionary::PodColumnColumns(_PodColumnColumns, _PodColumnColumns + COUNTOF(_PodColumnColumns));

const ColumnID Dictionary::_StashColumns[] = {StashID, StashHostID};
const ColumnIDs Dictionary::StashColumns(_StashColumns, _StashColumns + COUNTOF(_StashColumns));

const ColumnID Dictionary::_StashColumnColumns[] = {StashColumnStashID, StashColumnColumnID};
const ColumnIDs Dictionary::StashColumnColumns(_StashColumnColumns, _StashColumnColumns + COUNTOF(_StashColumnColumns));
